### DISCUSSION TITLE
Setting different print speeds for different filaments

### DISCUSSION THREAD

**Discussion by fox88x** [OrcaSlicer/OrcaSlicer#4130] [Q&A]
I would like to be able to create some print settings sets and associate them only with a specific filament so that selecting a different filament will not display these processes. I have tried three different approaches:

1) Modifying the .json files of the filament by adding speed definitions like "initial_layer_speed" = "70", etc., and deleting the respective definitions in the process .json file.
2) Adding something like "compatible_filament": ["Eryone PETG"] to the process file.
3) Adding a link to the process file in the filament file:
"process_list": [
    {
        "name": "0.20mm Eryone PETG",
        "sub_path": "process/0.20mm Eryone PETG.json"
     }
]

With all three approaches, I haven't achieved any results. Any ideas?

**Comment by aikiharp** [comment]
I like your idea. I currently have different process profiles for different filament types. For example, I have a 0.4mm nozzle, so I set up separate process profiles for for the filament types and layer heights I use (e.g., PETG 0.8mm, PLA 0.8mm, TPU 0.8mm) and the same for 0.20 for each type. It means maintaining a lot more process profiles and remembering to switch them when I use different filaments, but I like the results. I haven't dug into the .json files, but your idea of the process profile being linked to the filament type so it switches automatically would be nice. I've had to cancel a few prints because I forgot to switch one or the other, since I can't print PETG or TPU at the same speeds as PLA...

**Comment by git2212** [comment]
Orca seems to want to solve everything via "process" settings. I will opine that this is not the best approach. As an example, printing speeds (or at least overrides) are filament specific and not process specific whereas the number of walls, etc. are not filament dependent ... would be nice to have overrides at the filament level for that which is filament related...

**Comment by amstlubc** [comment]
I'd love to be able to have a base set of settings per printer/nozzle combo but be able to override in the filament the acceleration/speed.
